### PR TITLE
fix(entities): fix duplicate ids

### PR DESCRIPTION
**Pull request by yackinn**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/elf/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: 128

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/entities/src/lib/add.mutation.ts [hunk 1]
```diff
@@ -115,3 +114,19 @@ export function addEntitiesFifo<
...
   };
 }
+
+function throwIfEntityExists(
+  entities: any[],
+  idKey: string,
+  state: Record<any, any>,
+  entitiesKey: string
+) {
+  if (Array.isArray(entities)) {
```
Comment (NetanelBasal) [inline packages/entities/src/lib/add.mutation.ts:124]: It's always an array.

File: apps/ng/project.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
       "executor": "@angular-devkit/build-angular:browser",
       "outputs": ["{options.outputPath}"],
       "options": {
+        "buildOptimizer": false,
```
Comment (hakimio) [inline apps/ng/project.json:11]: Why would you disable optimizations for production build?

File: packages/entities/src/lib/add.mutation.ts [hunk 3]
```diff
@@ -40,15 +46,16 @@ export function addEntities<
     const { prepend = false, ref = defaultEntitiesRef } = options;
 
     const { entitiesKey, idsKey } = ref!;
+    const idKey = getIdKey<any>(context, ref);
 
     const asArray = coerceArray(entities);
 
     if (!asArray.length) return state;
+    if (process.env.NODE_ENV !== 'production') {
```
Comment (hakimio) [inline packages/entities/src/lib/add.mutation.ts:54]: Trying to access properties of undefined variable `process` in browser will throw.

**Comment by yackinn** [comment]
Sorry, it took a while to respond. We were trying some things and we decided to just use the functionality without using node environment variables.

**Comment by NetanelBasal** [comment]
@yackinn please update the docs

**Pull request merged**